discovery/manager.go

```diff
@@ -59,8 +60,18 @@ type poolKey struct {
 	provider string
 }
 
+type provider struct {
+	name   string
+	d      Discoverer
+	sets   []string
```

Comment:
it's not entirely clear to me what sets are supposed to be

Reply:
I used `sets` because `poolKey` has `setName` but I agree that the naming can be improved.

Reply:
okay so then `setName` is either scrape config `jobName` or the md5 checksum of the AlertmanagerConfig as JSON

Reply:
Correct